### PR TITLE
Fix wrong diff results

### PR DESCRIPTION
**Pull request by pgrivachev**
- Fixed cases when diff returned wrong result
- Fixed editing outside jinja in case error exists
- Zetasql-npm version fixed to 1.2.0 since latest release has some problems

### CODE REVIEW

File: server/src/test/diff/raw/4_lines_with_config.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+
+select id
+from `singular-vector-135519`.`dbt_ls_e2e_dataset`.`users`
```
Comment (fivetran-dmitryboykov) [inline server/src/test/diff/raw/4_lines_with_config.sql:4]: Looks like compiled, not raw.

File: e2e/src/helper.ts [hunk 2]
```diff
@@ -58,11 +58,18 @@ export async function activateAndWait(docUri: Uri): Promise<void> {
...
   console.log(JSON.stringify(e.contentChanges));
-  if (e.contentChanges.length === 1 && e.contentChanges[0].text === '') {
-    return;
-  }
 
   if (e.document.uri.path === 'Preview' && previewPromiseResolve) {
+    if (
+      // When we switch to a new document, the preview content is set to '' we skip this such events here
+      e.contentChanges.length === 1 &&
+      e.contentChanges[0].text === '' &&
```
Comment (fivetran-dmitryboykov) [inline e2e/src/helper.ts:66]: Please check, if it should be ' ' rather than '' ?

File: e2e/src/editing_outside_jinja.spec.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    const initialPreview = getPreviewText();
+
+    await replaceText('  )', '   ');
+    await replaceText('select u.id', 'select u.i');
+    assertThat(getPreviewText(), getMainEditorText());
+
+    await setTestContent(initialContent);
+    assertThat(getPreviewText(), initialPreview);
+  });
+});
```
Comment (fivetran-dmitryboykov) [inline e2e/src/editing_outside_jinja.spec.ts:19]: Recompilation did not start. But anyway when I change `await replaceText('select u.id', 'select u.i');` I receive different error. Is it right?

Comment (fivetran-dmitryboykov) [reply e2e/src/editing_outside_jinja.spec.ts:19]: After this change the next part of code is triggered:
```
async onCompilationFinished(compiledSql: string): Promise<void> {
    if (this.hasDbtError) {
      this.hasDbtError = false;
      this.onGlobalDbtErrorFixedEmitter.fire();
    }
```

**Review by fivetran-dmitryboykov — COMMENTED** [review]
Looks good. One more task seems to be fixed [https://fivetran.atlassian.net/browse/RD-225794](https://fivetran.atlassian.net/browse/RD-225794).

File: e2e/src/helper.ts (hunk 2 above)
Comment (pgrivachev) [reply e2e/src/helper.ts:66]: It should be ''. We receive such events when we loose focus from sql file for example and preview is cleared.

File: e2e/src/editing_outside_jinja.spec.ts (hunk 3 above)
Comment (pgrivachev) [reply e2e/src/editing_outside_jinja.spec.ts:19]: Could you please clarify what should I change to check this?

Comment (fivetran-dmitryboykov) [reply e2e/src/editing_outside_jinja.spec.ts:19]: Now global dbt error could be resolved after any change of non-jinja expression.

Comment (pgrivachev) [reply e2e/src/editing_outside_jinja.spec.ts:25]: Fixed it, please check.

**Review by fivetran-dmitryboykov — APPROVED** [review]

**Pull request merged**